### PR TITLE
make mimetype allow list configurable and add example config

### PR DESCRIPTION
**Pull request by wkloucek**
## Description
Since https://github.com/cs3org/reva/pull/2095 mime types must be on an allow list to be advertised on the `/app/list` endpoint. Additionally one can now add some more information about mime types and select a default application per mime type as requested in https://github.com/owncloud/ocis/issues/2563.

This PR adds a basic default configuration and the possibility to configure the mime types with a config file.

## Related Issue

-  Needs https://github.com/cs3org/reva/pull/2118 to work with multiple app providers
-  Needs https://github.com/cs3org/reva/pull/2095
- Fixes https://github.com/owncloud/ocis/issues/2563

## How Has This Been Tested?

- locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Core-API-Tests-ocis-storage-7</strong> failed. The build is cancelled...</summary>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Core-API-Tests-ocis-storage-5</strong> failed. The build is cancelled...</summary>

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,7 @@ node_modules/
 ocis/ocis
 ocis/cmd/ocis/__debug_bin
 ocis/cmd/ocis/config/
+config/
```
Comment (refs) [inline .gitignore:16]: 🙏  thanks

File: storage/pkg/command/gateway.go [hunk 2]
```diff
@@ -222,6 +228,118 @@ func rules(cfg *config.Config, logger log.Logger) map[string]map[string]interfac
...
+			Extension:   "zmd",
+			Name:        "Compressed markdown file",
+			Description: "Compressed markdown file",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+	}
+
+	if err := mapstructure.Decode(mimetypes, &m); err != nil {
+		return nil
```
Comment (refs) [inline storage/pkg/command/gateway.go:337]: that is a bit strange. How about an error return? 🙄

File: storage/pkg/command/gateway.go [hunk 3]
```diff
@@ -222,6 +228,118 @@ func rules(cfg *config.Config, logger log.Logger) map[string]map[string]interfac
 	}
 }
 
+func mimetypes(cfg *config.Config, logger log.Logger) (m map[string]map[string]string) {
```
Comment (refs) [inline storage/pkg/command/gateway.go:231]: just a style note, I wouldn't mix named returns with not-naked returns, it feels redundant to me. For style I'd prefer not named returns and explicit returns, even if that means 2 extra lines of variable declarations

**Review by refs — CHANGES_REQUESTED** [review]
Just style changes, other than that LGTM

File: storage/pkg/command/gateway.go (hunk 3 above)
Comment (wkloucek) [reply storage/pkg/command/gateway.go]: ok, removed the named returns

File: storage/pkg/command/gateway.go (hunk 2 above)
Comment (wkloucek) [reply storage/pkg/command/gateway.go:341]: I now added an error log, do you think that's sufficient?

**Review by refs — APPROVED** [review]

**Pull request merged**